scipy/sparse/_lil.py

```diff
@@ -287,6 +287,10 @@ def _mul_scalar(self, other):
     def __truediv__(self, other):           # self / other
         if isscalarlike(other):
             new = self.copy()
+            new_dtype = np.result_type(self.dtype, np.array(other).dtype)
+            if new_dtype != self.dtype:
+                new.data = new.data.astype(new_dtype)
+                new.dtype = new_dtype
```

Comment:
```suggestion
```
My mistake: this last step is not needed. Changing the dtype of self.data is enough to change the result of the property `new.dtype`.  

Thanks for putting this together!